torch/quantization/fake_quantize.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+from torch.nn import Module
+from .._jit_internal import weak_module
+from .observer import Observer, observer
+
+@weak_module
+class FakeQuantize(Module):
+    ''' Simulate the quantize and dequantize operations in training time.
+    Args:
+        `qconfig`: object that encodes configuration info for quantization
+        `observer_module`: Observer module that records stats of weights and
+        activations
+        `calcqparam`: A function that calculates quantization parameters
+        given the stats
+    '''
+
+    def __init__(self, qconfig, quant_min=-128, quant_max=127, quant_delay=0):
+        super(FakeQuantize, self).__init__()
+        self.observer = qconfig.activation()
+        self.qconfig = qconfig
+        self.quant_min = quant_min
+        self.quant_max = quant_max
+        self.quant_delay = quant_delay
+        self.iter = 1
```

Comment:
I have mixed feelings about it. Since in PyTorch user handles the training loop - do we need this handling inside the module? More idiomatic version seems to be something like this in user code:
```
train(loader, num_minibatches=1000)
enable_fake_quant(model)
train(loader)
```

Then the only flag we have to add to FakeQuantize is `enabled` and have an argument in constructor to initially disable it.

Reply:
that part is not yet fully designed, we can talk about that later, goal for this pr is just enabling the FakeQuantize module so that the eager mode API can use it.

Reply:
I agree, it is better to remove iter from fake_quant, instead just have:

 ```
     def enable_fake_quant(self):
        self.enable_fake_quant = True

   def forward(self,x):
            self.observer(X)
            if self.fake_quant_enable:
                self.scale, self.zero_point = self.calculate_qparams()
                X = torch.fake_quantize_per_tensor_affine(
                    X, self.scale.double(), self.zero_point.long(), self.quant_min,
                    self.quant_max)
              return X
```